e2e/helpers/wallet.helper.ts

```diff
@@ -157,4 +198,84 @@ export class WalletHelper {
 
         return { transaction, txid, blockhash };
     }
+
+    private createInputFromUtxo(utxo: UTXO, keyPair: BIP32Interface): any {
+        const input: any = {
+            hash: utxo.txid,
+            index: utxo.vout,
+        };
+
+        switch (utxo.addressType) {
+            case AddressType.P2SH_P2WPKH:
+                const p2wpkh = payments.p2wpkh({
+                    pubkey: keyPair.publicKey,
+                    network: networks.regtest,
+                });
+                const p2sh = payments.p2sh({
+                    redeem: p2wpkh,
+                    network: networks.regtest,
+                });
+                input.witnessUtxo = {
+                    script: p2sh.output,
+                    value: utxo.value,
+                };
+                input.redeemScript = p2sh.redeem.output;
+                break;
+            case AddressType.P2WPKH:
+                input.witnessUtxo = {
+                    script: payments.p2wpkh({
+                        pubkey: keyPair.publicKey,
+                        network: networks.regtest,
+                    }).output,
+                    value: utxo.value,
+                };
+                break;
+            case AddressType.P2PKH:
+                input.nonWitnessUtxo = Buffer.from(utxo.rawTx, 'hex');
+                break;
+            case AddressType.P2TR:
+                input.witnessUtxo = {
+                    script: payments.p2tr({
+                        internalPubkey: toXOnly(keyPair.publicKey),
+                        network: networks.regtest,
+                    }).output,
+                    value: utxo.value,
+                };
+                input.tapInternalKey = toXOnly(keyPair.publicKey);
+                break;
+        }
+
+        return input;
+    }
+}
+
+function getDerivationPath(addressType: AddressType, index: number): string {
+    switch (addressType) {
+        case AddressType.P2PKH:
+            return `m/44'/0'/0'/0/${index}`;
+        case AddressType.P2SH_P2WPKH:
+            return `m/49'/0'/0'/0/${index}`;
+        case AddressType.P2WPKH:
+            return `m/84'/0'/0'/0/${index}`;
+        case AddressType.P2TR:
+            return `m/86'/0'/0'/0/${index}`;
+        default:
+            throw new Error('Unsupported address type');
```

Comment:
for testnet we should `1'` as coin type
```suggestion
        case AddressType.P2PKH:
            return `m/44'/1'/0'/0/${index}`;
        case AddressType.P2SH_P2WPKH:
            return `m/49'/1'/0'/0/${index}`;
        case AddressType.P2WPKH:
            return `m/84'/1'/0'/0/${index}`;
        case AddressType.P2TR:
            return `m/86'/1'/0'/0/${index}`;
        default:
            throw new Error('Unsupported address type');
```